crates/storage/provider/src/post_state.rs

```diff
@@ -177,6 +181,43 @@ impl PostState {
         calculate_receipt_root_ref(self.receipts().iter().map(Into::into))
     }
 
+    /// Returns the hashed post state.
+    pub fn hash_state_slow(&self) -> HashedPostState {
+        let mut accounts = BTreeMap::default();
+        for (address, account) in self.accounts() {
+            accounts.insert(keccak256(address), *account);
+        }
+
+        let mut storages = BTreeMap::default();
+        for (address, storage) in self.storage() {
+            let mut hashed_storage = BTreeMap::default();
+            for (slot, value) in &storage.storage {
+                hashed_storage.insert(keccak256(H256(slot.to_be_bytes())), *value);
+            }
+            storages.insert(
+                keccak256(address),
+                HashedStorage { wiped: storage.wiped, storage: hashed_storage },
+            );
+        }
+
+        HashedPostState { accounts, storages }
+    }
+
+    /// Calculate the state root for this [PostState].
+    pub fn state_root_slow<'a, 'tx, TX: DbTx<'tx>>(
+        &self,
+        tx: &'a TX,
+    ) -> Result<H256, StateRootError> {
```

Comment:
this can be simplified
```suggestion
    pub fn state_root_slow<TX>(
        &self,
        tx: &TX,
    ) -> Result<H256, StateRootError>
    where for<'tx> TX: DbTx<'tx>
    {
```

Reply:
works for the method itself, but usage errors out with `implementation of `reth_db::transaction::DbTx` is not general enough`